common/build-style/python3-pep517.sh

```diff
@@ -19,7 +19,9 @@ do_check() {
 		testjobs="-n $XBPS_MAKEJOBS"
 	fi
 
-	local testdir="${wrksrc}/tmp/$(date +%s)"
+	local testdir="${wrksrc}/.xbps-testdir"
+	rm -rf "${testdir}"
```

Comment:
The benefit of cleaning up the test directory *after* we've used it is that we (attempt to) restore the source tree to the state it was in when the function was first called.

The benefit of cleaning up the directory before it would normally exist seems dubious. If we agree there is no need to clean up the test after a successful run, I think changing the directory name as you've done is sufficient.

Reply:
I just ran into the reason for the cleanup: attempting to re-run a check after a failure will cause `python3 -m installer` to fail unless the test directory is clean. Still, I think the idea of purging a (typically nonexistent) directory just to avoid issues with re-running tests is strange. At this point, I guess I'd rather see
```sh
local testdir="${wrksrc}/.xbps-testdir/$(date +s)"
```
and skip cleanup altogether. This should solve the problem of `pytest` trying to traverse into `tmp` without requiring a manual exclude, keeps our tests in a more-likely-unique subdirectory that should not conflict with any package contents, and avoid strange pre-clean steps. As a bonus, if you *do* run multiple tests to correct a failure, you can always do something like
```sh
diff -r .xbps-testdir/<failed> .xbps-testdir/<succeeeded>
```
after the fact to confirm what, if anything, has changed in the installed wheel.

Reply:
Ayyy, feels like I'm running in circles :laughing: This should be the Final.Final.Final.patch